### PR TITLE
adding Json-Schema for OpenLineage

### LINKED ISSUE #15: [PROPOSAL] Move the OpenLineage spec to JonSchema (from OpenAPI)
**Purpose:**
OpenAPI and JsonSchema are mostly the same model with a goal of OpenAPI to be a strict superset of JSONSchema in the near future. 
The goal is to make OpenLineage independent from the underlying protocol. OpenAPI assumes a REST endpoint which is not our intent here. This would still make possible to define an OpenLineage HTTP endpoint while decoupling the model and make it more obvious that it can be used with Kafka for example.

**Proposed implementation**
Convert the OpenAPI spec to a json schema.

### PR DESCRIPTION
**Pull request by julienledem**
resolves #15
Signed-off-by: Julien Le Dem <julien@apache.org>

### CODE REVIEW

File: spec/OpenLineage.json [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        },
+        {
+          "type": "object",
+          "properties": {
+            "run": {
+              "type": "object",
+              "properties": {
+                "runId": {
+                  "type": "string",
+                  "format": "uuid"
```
Comment (henneberger) [inline spec/OpenLineage.json:239]: type is string, not uuid

File: spec/OpenLineage.json [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+        {
+          "type": "object",
+          "properties": {
+            "type": {
+              "type": "string",
+              "example": "git"
+            },
+            "url": {
+              "type": "string",
+              "format": "uri"
```
Comment (henneberger) [inline spec/OpenLineage.json:309]: It is a bit confusing to me if this should be a url or uri.

File: spec/OpenLineage.json [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+      "type": "object"
+    },
+    "SourceCodeLocationJobFacet": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseFacet"
+        },
+        {
+          "type": "object",
+          "properties": {
```
Comment (henneberger) [inline spec/OpenLineage.json:302]: All of these properties are required

File: spec/OpenLineage.json [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+            }
+          },
+          "required": [
+            "description"
+          ]
+        }
+      ],
+      "type": "object"
+    },
+    "SourceCodeLocationJobFacet": {
```
Comment (henneberger) [inline spec/OpenLineage.json:295]: should these canned facets also extend additional properties for future backwards compatibility for changing facet versions?

File: spec/OpenLineage.json (hunk 1 above)
Comment (julienledem) [reply spec/OpenLineage.json:239]: thanks

File: spec/OpenLineage.json (hunk 4 above)
Comment (julienledem) [reply spec/OpenLineage.json:295]: that's a good point. Is it required?
Should all nested object types do the same?

File: spec/OpenLineage.json (hunk 3 above)
Comment (julienledem) [reply spec/OpenLineage.json:302]: thanks

File: spec/OpenLineage.json (hunk 2 above)
Comment (julienledem) [reply spec/OpenLineage.json:309]: a URL is a URI, so this is correct? https://tools.ietf.org/html/rfc3986#section-1.1.3
built-in formats have uri only. https://json-schema.org/understanding-json-schema/reference/string.html#resource-identifiers

File: spec/OpenLineage.json [hunk 5]
```diff
@@ -0,0 +1,426 @@
+{
+  "$schema": "http://json-schema.org/schema#",
```
Comment (wslulciuc) [inline spec/OpenLineage.json:2]: Should we add the [specific draft](https://json-schema.org/specification-links.html#published-drafts) of the standard we are using? For example: `http://json-schema.org/draft-07/schema#`

**Review by wslulciuc — APPROVED** [review]
Left a minor comment, but otherwise this looks great! And looks like there are plenty of [libraries](http://json-schema.org/implementations.html) for working with JSON Schema 🥇 💯

File: spec/OpenLineage.json (hunk 5 above)
Comment (julienledem) [reply spec/OpenLineage.json]: good point

**Pull request merged**